app/models/course.rb

```diff
@@ -117,6 +117,7 @@ def initialize_duplicate(duplicator, other)
     self.start_at += duplicator.time_shift
     self.end_at += duplicator.time_shift
     self.title = duplicator.new_course_title
+    self.registration_key = nil
```

Comment:
I think it should be `nil`, the lecturer should explicitly setup registration code enrolment if he wants.